aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/datamodeling/ConversionSchemas.java

```diff
@@ -158,6 +158,89 @@
 
     static final ConversionSchema DEFAULT = V2_COMPATIBLE;
 
+    /**
+     * A ConversionSchema builder that defaults to the {@link #V1} schema.
+     */
+    public static final Builder v1Builder(String name) {
+      return new Builder(name,
+              V1MarshallerSet.marshallers(), 
+              V1MarshallerSet.setMarshallers(), 
+              StandardUnmarshallerSet.unmarshallers(),
+              StandardUnmarshallerSet.setUnmarshallers());
+    }
+    
+    /**
+     * A ConversionSchema builder that defaults to the
+     * {@link #V2_COMPATIBLE} schema.
+     */
+    public static final Builder v2CompatibleBuilder(String name) {
+      return new Builder(name,
+              V2CompatibleMarshallerSet.marshallers(), 
+              V2CompatibleMarshallerSet.setMarshallers(), 
+              StandardUnmarshallerSet.unmarshallers(),
+              StandardUnmarshallerSet.setUnmarshallers());
+    }
+    
+    /**
+     * A ConversionSchema builder that defaults to the {@link #V2} schema.
+     */
+    public static final Builder v2Builder(String name) {
+      return new Builder(name,
+              V2MarshallerSet.marshallers(), 
+              V2MarshallerSet.setMarshallers(), 
+              StandardUnmarshallerSet.unmarshallers(),
+              StandardUnmarshallerSet.setUnmarshallers());
+    }
+    
+    public static class Builder {
+      
+      private final String name;
+      private final List<Pair<ArgumentMarshaller>> marshallers;
+      private final List<Pair<ArgumentMarshaller>> setMarshallers;
+      private final List<Pair<ArgumentUnmarshaller>> unmarshallers;
+      private final List<Pair<ArgumentUnmarshaller>> setUnmarshallers;
+      
+      Builder(String name,
+              List<Pair<ArgumentMarshaller>> marshallers,
+              List<Pair<ArgumentMarshaller>> setMarshallers,
+              List<Pair<ArgumentUnmarshaller>> unmarshallers,
+              List<Pair<ArgumentUnmarshaller>> setUnmarshallers) {
+        this.name = name;
+        this.marshallers = marshallers;
+        this.setMarshallers = setMarshallers;
+        this.unmarshallers = unmarshallers;
+        this.setUnmarshallers = setUnmarshallers;
+      }
+      
+      /**
+       * Adds marshaling of a type to the schema.
+       * Types are in LIFO order, so the last type added will be the first
+       * matched.
+       */
+      public void addType(Class<?> clazz, ArgumentMarshaller marshaller,
```

Comment:
I intuitively read pushX as appending to the end of a list, but this is adding to the front.
It is push_front and push_back in C++, but pushFrontType/pushTypeFront seems a bit of overkill. Unshift is the JavaScript convention, but that feels a bit wrong.
It could be addTypeFirst (to be consistent with LinkedList's naming)?
I don't feel too strongly. Do you have a strong preference?